Theme: Add new theme called "Base"

rxaviers opened the pull request:
Keep "Smoothness" and "UI Lightness" in the gallery, while the new
theme will be available as "Base" and it will be the default.

Ref https://github.com/jquery/jquery-ui/pull/1384

TODO:
- [x] Remove a temporary change below before landing.

rxaviers commented on config.json:
```diff
@@ -7,6 +7,11 @@
+			"dependsOn": "jQuery1.6+",
+			"label": "Experimental"
+		},
+		{
```
Note to self. This is a temporary change. Remove before landing.

rxaviers commented on themeroller.js:
```diff
@@ -82,6 +82,7 @@ Frontend.prototype = {
 
 	css: function( vars ) {
 		var theme = new ThemeRoller({
+			version: "1.12.0-pre",
```
Note to self. This is a temporary change. Remove before landing.

rxaviers commented:
Follow these instructions to run the server locally using the temporary changes:

1. Run `grunt prepare`. It will fail, because it won't find version `1a10f10cb0a39281e91eec5a53b5bb874497c629`.
1. Run `cd tmp/jquery-ui && git checkout --detach 20a636844961a1fb2de7a892ab28890091822e6a && git pull origin refs/pull/1384/head:pr/1384`.
1. Run `grunt prepare`. I must pass now.
1. Run `node server.js --console` and browse it.

rxaviers commented:
Themeroller displaying the new theme called "Base".

![screenshot from 2015-01-08 17 00 59](https://cloud.githubusercontent.com/assets/967155/5668611/0a71f900-9758-11e4-94ae-f301262877e8.png)

rxaviers commented:
jQuery UI 1.9.2, 1.10.4, 1.11.2, 1.12.0-pre packages with the new theme called "Base":
- https://gist.github.com/rxaviers/1f77d0a19806df75ff76

kborchers commented:
Didn't I see a better mockup somewhere with the rounded corners removed and a better font? This feels very similar to smoothness with just a slight color and maybe a font change when I look at it.

jzaefferer commented:
We changed the font-size in general, independent of this new theme. That doesn't seem to be included here, though. Rounded corners are still in the new theme, its just [smaller radii(?)](https://github.com/jquery/jquery-ui/pull/1384/files#diff-ff24a972bba1a2ecf79314fd9a5d048fR376). The smaller font-size might make the rounded corners more obvious.

One thing that looks very wrong on the screenshot is the overlay and shadow element demo. The overlay doesn't cover the background, the shadow also disappears in the middle.

rxaviers commented:
> We changed the font-size in general, independent of this new theme.

Please, can you point me to this change?

rxaviers commented:
> One thing that looks very wrong on the screenshot is the overlay and shadow element demo. The overlay doesn't cover the background, the shadow also disappears in the middle.

I'm assuming changes are going to be made on jquery/jquery-ui#1384, so I can later update this PR accordingly.

jzaefferer commented:
> Please, can you point me to this change?

https://github.com/jquery/jquery-ui/pull/1374

kborchers commented:
> Rounded corners are still in the new theme, its just smaller radii(?). The smaller font-size might make the rounded corners more obvious.

Let me rephrase my comment. Can we get rid of the rounded corners in this new base theme to make it feel more current?

jzaefferer commented:
You can check out the actual theme and discuss it at the original PR: https://github.com/jquery/jquery-ui/pull/1374

kborchers commented:
> You can check out the actual theme and discuss it at the original PR: jquery/jquery-ui#1374

I thought this was the Theme PR. The one you referenced is just the font size change so this would be the appropriate place to discuss the removal of the rounded corners. It doesn't seem like too much to ask for since most themes for current frameworks have removed rounded corners. They make it feel outdated no matter what you do to color or font.

kborchers commented:
Maybe you meant https://github.com/jquery/jquery-ui/pull/1384 which I see now. I will comment there.

jzaefferer commented:
Yes, sorry, that's the one I meant.

rxaviers commented:
This PR is now (temporarily) pointing to d2eb0c93cfc5fce158ac2af6ca7b0f50760d9b90, i.e., @jzaefferer's https://github.com/jquery/jquery-ui/pull/1420 updates.

I have it deployed locally. Please, just let me know if I can provide a screenshot of a particular place or perform any test and post here.

rxaviers commented:
Updated jQuery UI 1.9.2, 1.10.4, 1.11.2, 1.12.0-pre packages using the new "Base" theme (based on d2eb0c93cfc5fce158ac2af6ca7b0f50760d9b90). They're available for download on https://gist.github.com/rxaviers/edd9cb1ecdedb3e65b10.

Below is how I see the tooltip dropshadow on 1.12.0-pre package and on 1.11.2.

| 1.12.0-pre | 1.11.2 |
| --- | --- |
| ![screenshot from 2015-01-21 09 49 59](https://cloud.githubusercontent.com/assets/967155/5835756/583e544e-a153-11e4-8f2a-39f7679d1ba3.png) | ![screenshot from 2015-01-21 09 50 03](https://cloud.githubusercontent.com/assets/967155/5835757/59e181cc-a153-11e4-86d9-41d4d7f64f41.png) |

The Overlay shadow also changed. It looks better on 1.12.0-pre (although, the white stripe underneath it is still present). Using the Base theme on 1.11.2 or older packages is still problematic.

| 1.12.0-pre | 1.11.2 |
| --- | --- |
| ![screenshot from 2015-01-21 09 36 03 copy](https://cloud.githubusercontent.com/assets/967155/5835583/9e47328c-a151-11e4-9b87-e69f3dc3c944.png) | ![screenshot from 2015-01-21 09 35 59](https://cloud.githubusercontent.com/assets/967155/5835585/a0eacfda-a151-11e4-8d94-005fd89e9e46.png) |

rxaviers commented:
I've noticed a problem that happens since my first update. The initial Icons on 1.12-pre package are all duplicate / the same. In contrast, this doesn't happen on 1.11.2 (or older) using the Base theme.

| 1.12.0-pre | 1.11.2 |
| --- | --- |
| ![screenshot from 2015-01-21 09 36 03](https://cloud.githubusercontent.com/assets/967155/5835681/8da0ea44-a152-11e4-8417-8482234f59f6.png) | ![screenshot from 2015-01-21 09 35 59 copy](https://cloud.githubusercontent.com/assets/967155/5835682/90331700-a152-11e4-9cfe-2aef19592fba.png) |

rxaviers commented:
Jörn and I had a chat. Below is the outcome:
- [ ] Create a new `template/zip/index-1-12.html` with several style and markup fixes: (a) drop body font size override, (b) fix overlay markup, (c) fix caret icons (s/ui-icon-carat/ui-icon-caret/g), (d) fix menu markup (add div inside li's).
- [ ] Do not use an image when using the flat texture.
- [ ] Update "Base" theme variables according to @jzaefferer's tooltip shadow update.
- The white stripe of the overlay background is a local thing and should work just fine on the server (it'is due to bad image background generation).
- Should we change overlay background color.

Double check:
- [ ] On 1.9 and 1.10, accordion is not working properly in index.html demo page.

rxaviers commented:
I've edited my previous comment.

rxaviers commented:
Updated jQuery UI 1.9.2, 1.10.4, 1.11.2, 1.12.0-pre packages using the new "Base" theme (based on a2db6209ca77c901b0195683d330d1d8c935bb12). They're available for download on https://gist.github.com/rxaviers/5293bc717f283f5eaad7

rxaviers commented:
Updated jQuery UI 1.9.2, 1.10.4, 1.11.2, 1.12.0-pre packages using the new "Base" theme (based on 4371d59abd4a24275e0aaa4cae49a2fed591c9ae). They're available for download on https://gist.github.com/rxaviers/5293bc717f283f5eaad7

rxaviers commented:
> On 1.9 and 1.10, accordion is not working properly in index.html demo page.

Fixed this regression with dc11c77.

jzaefferer commented:
So with Jasper's last update and this fix, is everything addressed and we can merge both PRs to master? Or are we still missing something?

rxaviers commented:
It'd be good if @jaspermdegroot could make a final review (packages at https://gist.github.com/rxaviers/5293bc717f283f5eaad7). At least, browsing `index.html` and visualizing if the widgets with new Base theme are correct.

rxaviers commented:
To Whom It May Concern,

Based on the "we shouldn't bring back those comments/variables" decision (https://github.com/jquery/jquery-ui/pull/1436), TR is going to have dead controls for such variables until the baseline of the available releases for download is at least 1.12.0, where we could remove those controls from TR.

jzaefferer commented:
> TR is going to have dead controls for such variables until the baseline of the available releases for download is at least 1.12.0, when those controls could be removed from TR.

I'm okay with that. Specifically the shadow controls have probably seen very little use anyway.

Are we ready to land this? I suggest keeping the extra changes until we merge this and the UI theme PR to master, test on stage, then remove it until we release 1.12.

rxaviers commented:
> Are we ready to land this?

Yeap. I still suggest that @jaspermdegroot make a final review on all packages at https://gist.github.com/rxaviers/5293bc717f283f5eaad7.

> I suggest keeping the extra changes until we merge this and the UI theme PR to master, test on stage, then remove it until we release 1.12.

Ok.

jaspermdegroot commented:
I suppose the reason that all the changes in the default theme (except changes in ```ui-widget-shadow```) have been applied to previous versions of UI is that the ThemeRoller isn't version aware? Feels a bit wrong though to make changes in something that already has been released. The default theme in the ThemeRoller will be different from the one that you can download via the "Quick Downloads" links at the top of the Download Builder page.

1.12: The color for  ```ui-widget-shadow``` should be ```#666666``` instead of ```#dddddd```. I changed that in my last commit. See https://github.com/jquery/jquery-ui/commit/4371d59abd4a24275e0aaa4cae49a2fed591c9ae

Other than that it looks good!

jzaefferer commented:
> I suppose the reason that all the changes in the default theme (except changes in ui-widget-shadow) have been applied to previous versions of UI is that the ThemeRoller isn't version aware

That's wrong. Based on the discussion we had around this, it sounded like TR would use the `theme.css` file from the respective release, not just the newest. Was that wrong?

rxaviers commented:
I'm confused. I think @jaspermdegroot and @jzaefferer are talking about two different things. Correct me if I'm wrong.

I think @jaspermdegroot is talking about the fact the **Custom** Downloads for 1.12.0 and all the previous releases (1.11.2, 1.10.4 and 1.9.2) will include the **Base** theme by default from now on as decided prior to the creation of this PR. Although, the **Quick** Downloads will include **Base** for 1.12.0, but **UI Lightness** for the previous ones (given the fact those are existing old packages). I think Jasper has a valid point. But, the decision is up to you all.

@jzaefferer is talking about either of these two things: (a) the custom downloads should use the `theme.css` file from the respective release, which is true (please, just let me know if it's not); or (b)  the TR (the page, the web interface) should use the `theme.css` file from the respective release, which doesn't happen (TR, the interface, is not release-aware and it uses the latest released jQuery UI's `theme.css`).

jzaefferer commented:
Alright, I misunderstood that. Let's go ahead and merge this and the UI PR.

jzaefferer commented:
Upstream change landed in https://github.com/jquery/jquery-ui/commit/79c4fa1e7a09a3d28f0f8ff7a5b9f41a6d6e123d

jzaefferer commented:
Still need to integrate this into `jqueryui.com` to actually be able to test on stage.
